src/core/json/path.cc

```diff
@@ -291,6 +340,61 @@ auto Dfs::PerformStep(const PathSegment& segment, const JsonType& node, const Cb
   return {};
 }
 
+auto Dfs::MutateStep(const PathSegment& segment, const MutateCallback& cb, JsonType* node)
+    -> nonstd::expected<void, MatchStatus> {
+  switch (segment.type()) {
+    case SegmentType::IDENTIFIER: {
+      if (!node->is_object())
+        return make_unexpected(MISMATCH);
+
+      auto it = node->find(segment.identifier());
+      if (it != node->object_range().end()) {
+        bool erase = Mutate(cb, it->key(), &it->value());
+        if (erase) {
+          node->erase(it);
+        }
+      }
+    } break;
+    case SegmentType::INDEX: {
+      if (!node->is_array())
+        return make_unexpected(MISMATCH);
+      if (segment.index() >= node->size()) {
+        return make_unexpected(OUT_OF_BOUNDS);
+      }
+      bool erase = Mutate(cb, nullopt, &node[segment.index()]);
+      if (erase) {
```

Comment:
if you want to have a name for bool
if (bool erase = Mutate(cb, nullopt, &node[segment.index()]); erase) {

Reply:
removed erase